inc/add_site_ui/namespace.php

```diff
@@ -411,11 +436,12 @@ function handle_subdirectory( string $value ) : ?array {
 	$network_url = wp_parse_url( network_site_url() );
 	$network_host = $network_url['host'];
 
-	$path = trim( $value, '/' );
+	// Break out URL query parameters and hashes.
+	$path = trim( preg_split( '/[&\?#]/', $value )[0], '/' );
 
 	return [
-		'domain' => $network_host,
-		'path' => $path,
+		'domain' => idna_encode( $network_host ),
+		'path' => urlencode( $path ),
```

Comment:
Not sure why it was missing but just adding a slash prefix here so it's consistent with the other `handle_*` functions
```suggestion
		'path' => '/' . urlencode( $path ),
```